core/src/services/gcs/core.rs

```diff
@@ -112,9 +112,6 @@ impl GcsCore {
     }
 
     pub async fn sign<T>(&self, req: &mut Request<T>) -> Result<()> {
-        if self.allow_anonymous {
```

Comment:
Thanks so much for the quick fix. Could you also help with `load_token`? I think the logic should be to skip signing if `token.is_none()`. Additionally, `load_token` should correctly manage `allow_anonymous`.

Reply:
> skip signing if token.is_none()

I believe this is for the other PR which introduces the `token: Option<String>` field?

Reply:
> I believe this is for the other PR which introduces the `token: Option<String>` field?

I mean:

- `load_token()` should return `Result<Option<GoogleToken>>`
- if `self.token` is `Some()`, we should return it first `load_token`. (this can be done in the other PR)
- If `load_token()` returns `None`, `sign` should skip it.

Reply:
Ah yeah, I believe we're talking about the same things here.

I pushed https://github.com/apache/opendal/pull/4979/commits/54c1ae8b438bfa212c0f974fc9f072922d58c7c6 just as you wrote this. I believe this covers what you mention here.

Reply:
Great, thanks!